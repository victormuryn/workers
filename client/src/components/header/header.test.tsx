import React from 'react';
import {BrowserRouter as Router} from 'react-router-dom';
import renderer from 'react-test-renderer';

import {Provider} from 'react-redux';
import configureMockStore from 'redux-mock-store';

import Header from './header';

const mockStore = configureMockStore();

describe(`Header tests`, () => {
  it(`Header freelancer renders corrects`, () => {
    const store = mockStore({
      user: {
        username: `hellcaster`,
        accountType: `freelancer`,
      },
    });

    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Header/>
        </Router>
      </Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Header client renders corrects`, () => {
    const store = mockStore({
      user: {
        username: `yellowkiiid`,
        accountType: `client`,
      },
    });

    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Header/>
        </Router>
      </Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
