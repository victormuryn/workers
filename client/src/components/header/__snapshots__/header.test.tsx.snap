// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests Header client renders corrects 1`] = `
<nav
  className="navbar navbar-expand-md navbar-dark bg-dark"
>
  <div
    className="container"
  >
    <a
      className="inner-header__logo navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Workers
    </a>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        className="ms-auto me-0 navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          aria-current={null}
          className="ms-2 nav-link"
          data-rb-event-key={null}
          disabled={false}
          href="/projects"
          onClick={[Function]}
        >
          Проєкти
        </a>
        <a
          className="ms-2 nav-link"
          data-rb-event-key="#features"
          href="#features"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Features
        </a>
        <div
          className="ms-2 dropdown nav-item"
          onKeyDown={[Function]}
        >
          <a
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle nav-link"
            data-rb-event-key={null}
            href="#"
            id="header-profile"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
          >
            yellowkiiid
          </a>
        </div>
        <a
          className="ms-md-5 btn btn-success"
          disabled={false}
          href="/create"
          onClick={[Function]}
        >
          Створити проєкт
        </a>
      </div>
    </div>
  </div>
</nav>
`;

exports[`Header tests Header freelancer renders corrects 1`] = `
<nav
  className="navbar navbar-expand-md navbar-dark bg-dark"
>
  <div
    className="container"
  >
    <a
      className="inner-header__logo navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Workers
    </a>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        className="ms-auto me-0 navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          aria-current={null}
          className="ms-2 nav-link"
          data-rb-event-key={null}
          disabled={false}
          href="/projects"
          onClick={[Function]}
        >
          Проєкти
        </a>
        <a
          className="ms-2 nav-link"
          data-rb-event-key="#features"
          href="#features"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Features
        </a>
        <div
          className="ms-2 dropdown nav-item"
          onKeyDown={[Function]}
        >
          <a
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle nav-link"
            data-rb-event-key={null}
            href="#"
            id="header-profile"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
          >
            hellcaster
          </a>
        </div>
      </div>
    </div>
  </div>
</nav>
`;
